scalding-core/src/main/scala/com/twitter/scalding/StringUtility.scala

```diff
@@ -0,0 +1,39 @@
+package com.twitter.scalding
+
+object StringUtility {
+  // Scala's split function uses regex which is too slow thus we create this faster function
+  private def isTrailingWith(text: String, separator: String) = {
+    val result = text.foldRight((List[Boolean](), true)) {
+      case (c, (tillNow, isTrailingBefore)) =>
+        val trailing = isTrailingBefore && (c.toString == separator)
+        (tillNow ++ List(trailing), trailing)
```

Comment:
rather than putting on the back and calling reverse, why not: `trailing :: tilNow` which is O(1) and then you don't need the reverse?